server/src/test/java/org/opensearch/action/search/SearchRequestTests.java

```diff
@@ -74,6 +74,13 @@ protected SearchRequest createSearchRequest() throws IOException {
         );
     }
 
+    public void testClone() {
```

Comment:
Lets expand this test coverage to make sure that the complex data structures are also legit clones rather than just the top level object (e.g the intention of deep copy).  Based on the reference usages `fetchSource` seem like a clear candidate.

Reply:
Good point. Added another test for complex data structures.